### PR TITLE
fix: handle DYNAMIC_SERVER_USAGE in root layout for ISR pages

### PR DESCRIPTION
**Pull request by itstimwhite**
## Summary

Wraps the `headers()` call in the root layout (`apps/web/app/layout.tsx`) in a try-catch so public profile pages can use ISR (`revalidate = 3600`) without crashing the build. Next.js throws `DYNAMIC_SERVER_USAGE` when `headers()` is called during static/ISR rendering — the catch branch falls back to `nonce = undefined`, which is safe because public profile routes don't have CSP headers applied by the middleware.

## Test Coverage

- 1 new code path (catch branch → `nonce = undefined`). This path only triggers during Next.js static generation (ISR) and is not unit-testable without mocking Next.js internals.
- All 19 public profile routes verified passing via performance budget measurement (standalone production build).

## Pre-Landing Review

No issues found.

## Adversarial Review

Claude adversarial subagent found:
- **Broad catch** (informational): The bare `catch {}` catches any error, not just `DYNAMIC_SERVER_USAGE`. Accepted — `headers()` only throws this specific error during static rendering, and adding error-type checking adds complexity for a theoretical edge case.
- **CSP nonce disabled for ISR pages** (informational): Non-issue — public profile routes don't have CSP headers in the middleware, so the nonce was already unused for these routes.

## Test plan
- [x] Auth tests pass (421 passed, 1 pre-existing timeout — unrelated)
- [x] All 19 public profile routes pass performance budgets
- [x] Pre-push checks pass (biome, typecheck, lint)
- [x] Build succeeds with ISR pages

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Bug Fixes**
  * Public profile pages using Incremental Static Regeneration (ISR) with revalidate no longer cause build crashes.

* **Chores**
  * Release metadata updated and changelog consolidated for the 26.4.151.4 release.

### CODE REVIEW

File: apps/web/app/layout.tsx [hunk 1]
```diff
@@ -139,7 +139,14 @@ export default async function RootLayout({
...
 }>) {
-  const nonce = (await headers()).get(SCRIPT_NONCE_HEADER) ?? undefined;
+  // headers() throws DYNAMIC_SERVER_USAGE during ISR/static rendering (e.g.
+  // public profile pages with revalidate). Fall back to no nonce in that case.
+  let nonce: string | undefined;
+  try {
+    nonce = (await headers()).get(SCRIPT_NONCE_HEADER) ?? undefined;
+  } catch {
+    nonce = undefined;
+  }
```
[the remark this answers is not in the document]
Comment (itstimwhite) [reply apps/web/app/layout.tsx:150]: Acknowledged — the dynamic server usage handling in the root layout needs careful scoping to avoid breaking ISR pages.

[the remark this answers is not in the document]
Comment (itstimwhite) [reply apps/web/app/layout.tsx:150]: Valid concern. Will ensure the DYNAMIC_SERVER_USAGE catch is scoped correctly and doesn't inadvertently disable caching for static routes.
